src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1439,574 +1687,35 @@ internal void Stop()
         /// </summary>
         internal void Done()
         {
-            if (_inputThread != null)
+            // we allow call Done() multiply times.
+            // For example one time from Process() code path, 
+            // when we detect that process already finished
+            // and once from End() code path.
+            if (_pipeline != null)
             {
-                _inputThread.Join();
+                _pipeline.End();
+                _pipeline.Dispose();
+                _pipeline = null;
             }
-        }
 
-        /// <summary>
-        /// Thread procedure for writing data to the child process...
-        /// </summary>
-        private void WriterThreadProc()
-        {
-            try
+            // streamWriter is present, only if we call Start method
+            if (_streamWriter != null)
             {
-                if (_inputFormat == NativeCommandIOFormat.Text)
+                try
                 {
-                    WriteTextInput();
+                    _streamWriter.Dispose();
                 }
-                else
+                catch (IOException)
                 {
-                    WriteXmlInput();
+                    // on unix, if process is already finished attempt to dispose it will
+                    // lead to "Broken pipe" exception.
+                    // we are ignoring it here
```

Comment:
I dig into that and as far as I see, that will hang the process: there is no way to tell it that all input was present and close the stream, if process expects input.